lib/smwc_bot/search.ex

```diff
@@ -0,0 +1,58 @@
+defmodule SMWCBot.Search do
+  @moduledoc """
+  Search SMWC for shit.
+  """
+
+  require Logger
+
+  @doc """
+  Search for hack.
+  """
+  @spec for({String.t(), String.t()}) :: {text :: String.t(), href :: String.t()} | nil
+  def for({hack, waiting}) do
+    base_uri = base_uri(waiting)
+    filter = URI.encode_query(%{"f[name]" => hack})
+    search_uri = "#{base_uri}#{filter}"
+
+    Logger.debug("Uri = #{search_uri}")
+
+    result_page = HTTPoison.get!("#{base_uri}#{filter}")
```

Comment:
In the future could handle this differently so that a message could be returned to the chat use when this fails